mutt/charset.c

```diff
@@ -1019,42 +1016,36 @@ void mutt_ch_set_charset(const char *charset)
 /**
  * mutt_ch_choose - Figure the best charset to encode a string
  * @param[in] fromcode Original charset of the string
- * @param[in] charsets Colon-separated list of potential charsets to use
+ * @param[in] charsets Slist of potential charsets to use
  * @param[in] u        String to encode
  * @param[in] ulen     Length of the string to encode
  * @param[out] d       If not NULL, point it to the converted string
  * @param[out] dlen    If not NULL, point it to the length of the d string
  * @retval ptr  Best performing charset
  * @retval NULL None could be found
  */
-char *mutt_ch_choose(const char *fromcode, const char *charsets, const char *u,
+char *mutt_ch_choose(const char *fromcode, const struct Slist *charsets, const char *u,
                      size_t ulen, char **d, size_t *dlen)
 {
-  if (!fromcode)
+  if (!fromcode || !charsets)
     return NULL;
 
   char *e = NULL, *tocode = NULL;
   size_t elen = 0, bestn = 0;
-  const char *q = NULL;
+  struct ListNode *np = NULL;
 
-  for (const char *p = charsets; p; p = q ? q + 1 : 0)
+  STAILQ_FOREACH(np, &charsets->head, entries)
   {
-    q = strchr(p, ':');
-
-    size_t n = q ? q - p : strlen(p);
+    char *p = (char *) np->data;
+    size_t n = strlen(p);
     if (n == 0)
```

Comment:
again, not your fault, but can we have some more descriptive variable names, rather than `p` and `n`